torch/linalg/__init__.py

```diff
@@ -339,6 +339,10 @@
 
 .. note:: For CUDA inputs, this function synchronizes that device with the CPU.
 
+.. warning:: This function assumes that :attr:`A` is `diagonalizable`_ (e.g. when all the
+             eigenvalues are different). If it is not diagonalizable, the returned
+             eigenvalues will be correct but :math:`A \neq V \operatorname{diag}(\Lambda)V^{-1}`.
```

Comment:
I am not sure that `diagonazible` implies unique eigenvalues, more like for repeated eigenvalues we must have that the geometric multiplicity is equal to the algebraic multiplicity, or am I missing something?

Reply:
e.g. means "for example", meaning that the implication goes "different eigenvalues" => diagonalisable. It is indeed not an if and only if for the reasons you point out.